File: src/SOP8/Egovframework/Base/base-history-ibll/src/main/java/egovframework/base/history/ibll/models/SOP/SOPHistoryComponentData.java
package egovframework.base.history.ibll.models.sop;

import java.util.ArrayList;
import java.util.List;

public class SOPHistoryComponentData {

    private int actionStepHistoryNo = -1;
    private int componentHistoryNo = -1;
    private int componentNo = -1;
    private int componentType = -1;
    private String title = "";
    private List<String> teamList = new ArrayList<>();
    private String time = "";
    private String status = "";
    private Integer userNo = null;
    private String userName = "";
    private String completion = "확인";
    private List<ComponentHistoryDetailData> missionDatas = new ArrayList<>();

    public int getActionStepHistoryNo() {
        return actionStepHistoryNo;
    }

    public void setActionStepHistoryNo(int actionStepHistoryNo) {
        this.actionStepHistoryNo = actionStepHistoryNo;
    }

    public int getComponentHistoryNo() {
        return componentHistoryNo;
    }

    public void setComponentHistoryNo(int componentHistoryNo) {
        this.componentHistoryNo = componentHistoryNo;
    }

    public int getComponentNo() {
        return componentNo;
    }

    public void setComponentNo(int componentNo) {
        this.componentNo = componentNo;
    }

    public int getComponentType() {
        return componentType;
    }

    public void setComponentType(int componentType) {
        this.componentType = componentType;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public List<String> getTeamList() {
        return teamList;
    }

    public void setTeamList(List<String> teamList) {
        this.teamList = teamList;
    }

    public String getTime() {
        return time;
    }

    public void setTime(String time) {
        this.time = time;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Integer getUserNo() {
        return userNo;
    }

    public void setUserNo(Integer userNo) {
        this.userNo = userNo;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getCompletion() {
        return completion;
    }

    public void setCompletion(String completion) {
        this.completion = completion;
    }

    public List<ComponentHistoryDetailData> getMissionDatas() {
        return missionDatas;
    }

    public void setMissionDatas(List<ComponentHistoryDetailData> missionDatas) {
        this.missionDatas = missionDatas;
    }

    // Inner class
    public static class ComponentHistoryDetailData {

        private int dataIndex = -1;
        private String sectionName = "";
        private String missionText = "";
        private String time = "";
        private String completion = "미완료";

        public int getDataIndex() {
            return dataIndex;
        }

        public void setDataIndex(int dataIndex) {
            this.dataIndex = dataIndex;
        }

        public String getSectionName() {
            return sectionName;
        }

        public void setSectionName(String sectionName) {
            this.sectionName = sectionName;
        }

        public String getMissionText() {
            return missionText;
        }

        public void setMissionText(String missionText) {
            this.missionText = missionText;
        }

        public String getTime() {
            return time;
        }

        public void setTime(String time) {
            this.time = time;
        }

        public String getCompletion() {
            return completion;
        }

        public void setCompletion(String completion) {
            this.completion = completion;
        }
    }
}

File: src/SOP8/Base/History/Base.History.IBLL/Models/SOP/SOPHistoryComponentData.cs
using System.Collections.Generic;

namespace Base.History.IBLL.Models.SOP
{
    public class SOPHistoryComponentData
    {
        private int m_nActionStepHistoryNo = -1;
        private int m_nComponentHistoryNo = -1;
        private int m_nComponentNo = -1;
        private int m_nComponentType = -1;
        private string m_strTitle = "";
        private List<string> m_teamList = new List<string>();
        private string m_strTime = "";
        private string m_strStatus = "";
        private int? m_userNo = null;
        private string m_strUserName = "";
        private string m_strCompletion = "확인";
        private List<ComponentHistoryDetailData> m_missionDatas = new List<ComponentHistoryDetailData>();

        public int ActionStepHistoryNo
        {
            get { return m_nActionStepHistoryNo; }
            set { m_nActionStepHistoryNo = value; }
        }
        public int ComponentHistoryNo
        {
            get { return m_nComponentHistoryNo; }
            set { m_nComponentHistoryNo = value; }
        }
        public int ComponentNo
        {
            get { return m_nComponentNo; }
            set { m_nComponentNo = value; }
        }
        public int ComponentType
        {
            get { return m_nComponentType; }
            set { m_nComponentType = value; }
        }
        public string Title
        {
            get { return m_strTitle; }
            set { m_strTitle = value; }
        }
        public List<string> TeamList
        {
            get { return m_teamList; }
            set { m_teamList = value; }
        }
        public string Time
        {
            get { return m_strTime; }
            set { m_strTime = value; }
        }
        public string Status
        {
            get { return m_strStatus; }
            set { m_strStatus = value; }
        }
        public int? UserNo
        {
            get { return m_userNo; }
            set { m_userNo = value; }
        }
        public string UserName
        {
            get { return m_strUserName; }
            set { m_strUserName = value; }
        }
        public string Completion
        {
            get { return m_strCompletion; }
            set { m_strCompletion = value; }
        }
        public List<ComponentHistoryDetailData> MissionDatas
        {
            get { return m_missionDatas; }
            set { m_missionDatas = value; }
        }
    }

    public class ComponentHistoryDetailData
    {
        private int m_nDataIndex = -1;
        private string m_strSectionName = "";
        private string m_strMissionText = "";
        private string m_strTime = "";
        private string m_strCompletion = "미완료";

        public int DataIndex
        {
            get { return m_nDataIndex; }
            set { m_nDataIndex = value; }
        }
        public string SectionName
        {
            get { return m_strSectionName; }
            set { m_strSectionName = value; }
        }
        public string MissionText
        {
            get { return m_strMissionText; }
            set { m_strMissionText = value; }
        }
        public string Completion
        {
            get { return m_strCompletion; }
            set { m_strCompletion = value; }
        }
        public string Time
        {
            get { return m_strTime; }
            set { m_strTime = value; }
        }
    }
}
